Source/ScaleStructureController/TransparentDropDown.h

```diff
@@ -0,0 +1,133 @@
+/*
+  ==============================================================================
+
+    ComboBoxLookAndFeel.h
+    Created: 13 May 2020 9:13:28pm
+    Author:  Vincenzo
+
+  ==============================================================================
+*/
+
+#pragma once
+#include <JuceHeader.h>
+
+class TransparentDropDown : public LookAndFeel_V4
+{
+	Colour defaultColour = Colours::darkgrey;
+	Colour& baseColour;
+
+public:
+
+	TransparentDropDown()
+		: baseColour(defaultColour) {};
+
+	~TransparentDropDown() override {};
+
+	void setBaseColour(Colour& baseColourIn)
+	{
+		baseColour = baseColourIn;
+	}
+
+	// ComboBox Overrides
+
+	void drawComboBox(
```

Comment:
Is this JUCE’s concept for custom drawing of combos boxes? I have been looking for this kind of thing in the past (in vain so far - good to see this!). Doesn‘t appear that obvious to me to have the drawing functionality in the LokkAndFeel component instead of in the Combobox itself - but if that’s the design of JUCE, it’s OK.

Reply:
Yes this is JUCE's concept for custom component drawing. I agree it is a bit odd, but I think the idea is it can be more versatile and allow for more reusable code. For example, I can apply this same "TransparentDropDown" LookAndFeel to other components/classes than just ComboBox (although in this case it would only make sense for certain ones like PopupMenu). Here's the doc I would recommend to understand this better:
https://docs.juce.com/master/classLookAndFeel__V4.html

Although even with that reference, I had to do some trial and error and simple things like centering the dropdown menus around a component aren't trivial for menus that vary in size.

Reply:
With LookAndFeel_V4, you only have to override the functions you want to use. The previous LookAndFeel classes have some pure virtual functions.